### ISSUE TITLE
予約機能を作成する

### ISSUE THREAD

**Issue by y-tomimoto** [y-tomimoto/TAKITATE#8]
アプリから予約するための検索欄を作る。
検索してヒットした店舗をタップして、予約機能を作成する。

Search Viewを採用する。
> https://developer.android.com/training/search/setup?hl=ja

**Comment by y-tomimoto** [comment]
# meta-data 
> https://developer.android.com/guide/topics/manifest/meta-data-element?hl=ja

**Comment by y-tomimoto** [comment]
# Missing accessibility label について
idを指定してもwarningが維持されるので、下記でwarningを消す
> https://dev.classmethod.jp/articles/android-ignore-build-warning/
> https://stackoverflow.com/questions/52868488/missing-accessibility-label-in-android-studio-warning/52868733

**Comment by y-tomimoto** [comment]
# DBを追加したので、検索によって保存した値を取得する
- [x] まずDBから指定したITEMで検索し、取得した値を表示する
- [x] そのロジックを検索欄に当てはめる。 

> http://olee46.hatenablog.com/entry/2017/05/31/191657

- [x] 遷移先のレイアウトで検索を行う

ファイルが存在するのにR.id...の名前解決ができない場合
> https://www.quora.com/How-do-I-solve-cannot-resolve-symbol-r-in-android-studio

**Comment by y-tomimoto** [comment]
#  導線について
1. 予約する店舗を選ぶ(QR or 検索 or カテゴリ or ... )
2. 店舗の予約画面に遷移(このとき、店舗ごとの会員ステータスを用意する画面に遷移させる？)
3. 店舗の空き時間データを取得し、結果に反映する　(スキーマは別のissueで管理する)
4. Userの予約可能ステータスを確認（脱毛なら一度の来院に付き1予約とか）
5. 予約可能時間をカレンダーで表示する
6. 予約可能時間をタップする（タップ順に希望日を選択する）
7.  予約完了表示をする。実態は事業者側でpending扱いになっており、キャンセルすると通知が行く
8. （続けて予約する）をトースト表示。タップされない場合はhomeに戻る。

**Comment by y-tomimoto** [comment]
# 1. 予約する店舗を選ぶ(QR or 検索 or カテゴリ or ... )
実際のユースケースを考えると、
1. 店舗でinstall => QRを読み取り予約（このとき、QRから流入した店舗はSharedに保存しておく）
2. 次回は、（最近の店舗）欄から予約できるようにする。
ということで、まずはQRでの予約機能から追加していく。
# QRを読み取り、特定の店舗の予約画面に遷移する
- [ ] QRから読み取った値を、Activityに渡す
> https://qiita.com/hoshiume11/items/0f496fe80df84875c132
- [ ] 受け取った値を元に遷移先の店舗のUIのもとになるデータを取得する
- [ ] 店舗画面を表示する

**Comment by y-tomimoto** [comment]
かなり活発な印象を受ける。

![image](https://user-images.githubusercontent.com/31181316/83335237-562a1e00-a2e6-11ea-96a8-769ea4491581.png)

**Comment by y-tomimoto** [comment]
openCVって手もある。できるだけユーザビリティを重視したいのでもう少し調査。
> https://www.tcmobile.jp/dev_blog/programming/android%E3%81%AEqr%E3%82%B3%E3%83%BC%E3%83%89%E3%83%AA%E3%83%BC%E3%83%80%E3%83%BC%E3%81%AFzxing%E3%82%88%E3%82%8Aopencv4%E3%81%8C%E7%B2%BE%E5%BA%A6%E3%81%84%E3%81%84/

**Comment by y-tomimoto** [comment]
google-visionはintentを介することができないみたいな書き方だけど、そうなのか？
精度はgoogle-visionが良さそうなので、先にgoogle-visionを試す。
> https://medium.com/@lkumar.sakare/zxing-vs-google-vision-fc3be8d83ace

**Comment by y-tomimoto** [comment]
これ。ドキュメントが結構甘いな。
> https://developers.google.com/vision/android/barcodes-overview

**Comment by y-tomimoto** [comment]
こっちがチュートリアルか
> https://codelabs.developers.google.com/codelabs/bar-codes/#2

**Comment by y-tomimoto** [comment]
少し古いように見える。
![image](https://user-images.githubusercontent.com/31181316/83467195-2d578380-a4b4-11ea-8cd4-57133f38e4a9.png)

やはりcompileは非推奨。
> https://qiita.com/opengl-8080/items/6ad642e0b016465891de

**Comment by y-tomimoto** [comment]
こちらを参照する。
> https://developers.google.com/android/guides/setup

**Comment by y-tomimoto** [comment]
おや、一覧にない、、、どこかに内包されているか
![image](https://user-images.githubusercontent.com/31181316/83467720-b9b67600-a4b5-11ea-9afe-dfc7023add1f.png)

**Comment by y-tomimoto** [comment]
vision内に含まれているのか。
![image](https://user-images.githubusercontent.com/31181316/83467848-0b5f0080-a4b6-11ea-9716-e37500caaa78.png)

**Comment by y-tomimoto** [comment]
使い方同じならいいが、、、
> https://developers.google.com/android/reference/com/google/android/gms/vision/barcode/BarcodeDetector?hl=ja

**Comment by y-tomimoto** [comment]
build.gradleに追加
![image](https://user-images.githubusercontent.com/31181316/83468048-9f30cc80-a4b6-11ea-8aee-bf4ef5af44cd.png)

**Comment by y-tomimoto** [comment]
そういえば、携帯のカメラでQRを撮って、app or web の予約ページを開くことはあっても、app内でQRをとって予約することってないかもな。

**Comment by y-tomimoto** [comment]
一旦別のissueで対応する。

**Comment by y-tomimoto** [comment]
まず予約画面から作成する。

**Comment by y-tomimoto** [comment]
ピボットのためclose
